@here/harp-geoutils/lib/tiling/SubTiles.ts

```diff
@@ -7,78 +7,45 @@
 import { TileKey } from "./TileKey";
 
 export class SubTiles implements Iterable<TileKey> {
-    private m_tileKey: TileKey;
-    private m_level: number;
-    private m_count: number;
-    private m_mask: number;
-    private m_shift: number;
-
-    constructor(tileKey: TileKey, level: number, mask: number) {
-        this.m_tileKey = tileKey;
-        this.m_level = level;
-        // tslint:disable:no-bitwise
-        this.m_count = 1 << (level << 1);
-        this.m_mask = mask;
-        this.m_shift = level > 2 ? (level - 2) << 1 : 0;
-        // tslint:enable:no-bitwise
-    }
-
-    get level(): number {
-        return this.m_level;
-    }
-    get tileKey(): TileKey {
-        return this.m_tileKey;
-    }
+    constructor(public tileKey: TileKey, public sizeX: number, public sizeY: number) {}
```

Comment:
SubTiles was really working only on a quad-tree, so yeah it is not working. Still, it would be nice if we could keep using for quad trees.

Reply:
done. and i think, z-curve method would be a bit cheaper, as it uses just one loop and no multiplications.